### ISSUE TITLE
Type check with @types/react 18 failing

### ISSUE THREAD

**Issue by bryanjtc** [Semantic-Org/Semantic-UI-React#4352]
### Steps
Just install @types/react, @types/react-dom v18 and typescript and update react and react-dom to v18 in any project using semantic-ui-react

### Expected Result
No error

### Actual Result
Type checking throws this errors:
```ts
JSX element type 'Header' does not have any construct or call signatures
```
The same error happens with Segment, Grid, List and Image

### Version
2.1.2

### Testcase

[Fork, update, and replace this pen to show the bug]:

[Sandbox example](https://codesandbox.io/s/semantic-ui-react-forked-oyzfb2?file=/index.tsx)

![image](https://user-images.githubusercontent.com/49354825/162587018-64c16647-e1f8-4b2e-af7e-d07cfd6d5493.png)

I think this error is happening because many elements extend from the type `StatelessComponent`.
This has breen deprecated.
![image](https://user-images.githubusercontent.com/49354825/162586994-dc429f88-2c5c-456a-ae0e-fa0bbaeca6e8.png)

**Comment by casperOne** [comment]
Can confirm, currently seeing it with any component that has child elements after upgrading to react types 18.

**Comment by jchook** [comment]
It looks like semantic-ui-react does not support React 18.x yet. 

The [package.json](https://github.com/Semantic-Org/Semantic-UI-React/blob/9dd0596a8ab39e6c1d47d5630ddb878ec31a7d7a/package.json#L190) specifies `react@"^16.8.0 || ^17.0.0"` as a peer dependency.

**Comment by layershifter** [comment]
The fix was released in `semantic-ui-react@2.1.3`.
